app/controllers/events_controller.rb

```diff
@@ -0,0 +1,48 @@
+class EventsController < ApplicationController
+  # To-Do : Use Pundit for better authorization
+	before_action :authenticate_user!, except: [:index, :home,  :show]
+  before_action :load_event, only: [:show, :update, :destroy]
+
+	def index
+		@events = Event.all
+    #Implement searching and pagination using Pagy gem
+	end
+
+	def create
+		@event = Event.new(event_params)
+		@event.user_id = current_user.id
+
+		if @event.save
+			render notice: t("resource.created", { resource_name: "Event" })
+		end
+	end
+
+	def show
+		@event_bookings = @event.event_bookings
+
+		if params[:booking_id]
+			@booking_confirmed = current_user.event_bookings.find(params[:booking_id])
+		end
+	end
+
+	def update
+		if @event.update_attributes(event_params)
+			render notice: t("resource.updated", { resource_name: "Event" })
+		end
```

Comment:
Delegate further logic in separate service to handle event changes

```ruby
module Event
  class UpdateService 
      attr_reader :event, :user, :response, :event_params

    def initialize(event, user, event_params)
      @event = event
      @user = user
      @event_params = event_params
    end

    def process!
      if @event.user == user && @event.update(event_params)
         notify_visitors
         response = { success: true }
      else
        response = { success: false }
        response[:error] = I18n.t("unauthorized") if @event.user != user
      end
    end

    private

      def notify_visitors
        Visitors::NotificationService.new(event:).process
      end
  end
end

```

`Visitors::NotificationService` handles mail, mobile notifications, or integration effects actions on Event changes like venue, dates etc.